aquifer-git.js

```diff
@@ -100,185 +98,96 @@ module.exports = (Aquifer, AquiferGitConfig) => {
-          let dest = path.join(destPath, link.dest);
-          fs.removeSync(dest);
-          fs.copySync(src, dest, {clobber: true});
-        });
-      })
-
-      // Add all files to the index.
-      .then(() => {
-        Aquifer.console.log('Adding all files to the index...', 'status');
-        return repo.index();
-      })
-      .then((index_) => {
-        index = index_;
-        return index.removeAll();
-      })
-      .then(() => {
-        index.write();
-      })
-      .then(() => {
-        return index.addAll();
-      })
-      .then(() => {
-        index.write();
-      })
-
-      // Commit changes.
-      .then(() => {
-        let signature;
-
-        Aquifer.console.log('Commit changes...', 'status');
-
-        if (options.name && options.email) {
-          // Use custom signature.
-          signature = git.Signature.now(options.name, options.email);
-        }
-        else {
-          // Use default signature.
-          signature = git.Signature['default'](repo);
-        }
-
-        return repo.createCommitOnHead(['.'], signature, signature, options.message);
-      })
-
-      // Push changes.
-      .then(() => {
-        Aquifer.console.log('Pushing changes to origin/' + options.branch + '...', 'status');
-        return repo.getRemote('origin');
-      })
-      .then((remote) => {
-        let ref   = 'refs/heads/' + options.branch;
-        let refs  = [ref + ':' + ref];
-        let pushOptions = {
-          callbacks: {
-            certificateCheck: () => { return 1; },
-            credentials: (url, userName) => {
-              return git.Cred.sshKeyFromAgent(userName);
-            }
-          }
-        };
-
-        return remote.push(refs, pushOptions);
-      })
-
-      // Remove the destination path.
-      .then(() => {
-        Aquifer.console.log('Removing the ' + destPath + ' directory...', 'status');
-        fs.removeSync(destPath);
-      })
-
-      // Success!
-      .then(() => {
-        Aquifer.console.log('The site has been successfully deployed!', 'success');
-      })
-
-      // Catch any errors.
-      .catch((err) => {
-        callback(err);
+      // Create instance of build object.
+      build = new Aquifer.api.build(Aquifer);
+      return build.create(buildPath, buildOptions);
+    })
+
+    // Copy over additional deployment files.
+    .then(() => {
+      Aquifer.console.log('Copying deployment files...', 'status');
+      options.deploymentFiles.forEach(function (link) {
+        let src = path.join(Aquifer.projectDir, link.src);
+        let dest = path.join(destPath, link.dest);
+        fs.removeSync(dest);
+        fs.copySync(src, dest, {clobber: true});
       });
+    })
+
+    // Add all files to the index.
+    .then(() => {
+      Aquifer.console.log('Clearing the index...', 'status');
+      console.log(path.join(Aquifer.projectDir, destPath));
+      fs.removeSync(path.join(Aquifer.projectDir, destPath, '.git/index'));
```

Comment:
@chasingmaxwell I was occasionally getting the error during `git add -A`:
> fatal: multiple stage entries for merged file 'path/to/file'
This seemed to fix it.